### ISSUE TITLE
scanCount is uint16_t but scanNetworks returns only int8_t

### ISSUE THREAD

**Issue by theproxy** [espressif/arduino-esp32#1496]
### Hardware:
Board:							ESP32 Dev Module
IDE name:						Currently using Platform.io with ESP32
Flash Frequency:					?40Mhz?
Upload Speed:						?921600?
SYSTEM: ESP32 240MHz 288KB RAM (1.25MB Flash)

### Description:
When scanning networks with many many access points, scanNetworks returns negative number (not -1 or -2) and you can still read the output of scanNetworks but now have to deal with overflowed number returned from scanNetworks.

**Comment by theproxy** [comment]
in file https://github.com/espressif/arduino-esp32/blob/master/libraries/WiFi/src/WiFiScan.cpp

**Comment by lbernstone** [comment]
libraries/WiFi/src/WiFiType.h:#define WIFI_SCAN_RUNNING   (-1)
libraries/WiFi/src/WiFiType.h:#define WIFI_SCAN_FAILED    (-2)

**Comment by theproxy** [comment]
Hi thank you for the reply. I saw the -1 and -2 in the code. What I am seeing (still gathering data on this) is that I am getting a number like -67 or so. I have structured my code that if I see scanNetworks return < -2, then I can ask for at least 127 WiFi.SSID(i) and I get valid data from that call up through that many indices.

It's like walking a tightrope though because I am guessing that there is that much information available in the WiFI data structure, but I know it's a hack to get that data by blindly hardcoding the for loop.
